### ISSUE TITLE
Is there a way to put a border to the shape?

### ISSUE THREAD

**Issue by jsmedmar** [amueller/word_cloud#46]

**Comment by amueller** [comment]
Sorry, I don't understand the question.

**Comment by JuanSanMedina** [comment]
Sorry, it was late and I was tired. I was looking to do something like this:

![falsi](https://cloud.githubusercontent.com/assets/8843150/6763023/f05dae24-cf42-11e4-8afe-655fc60800fb.png)

I did the border line with illustrator, and I was wondering if that could be done from word_cloud

**Comment by amueller** [comment]
Not directly, but it should be very easy in python. You already provided the mask, so you can just call some morphology function on the mask to get the border, and then add the border and the word cloud image.

**Comment by amueller** [comment]
Actually that might be a nice addition if you like to code it up.

**Comment by JuanSanMedina** [comment]
@amueller ok, give me a try.

**Comment by gisellecory** [comment]
Hi - Did anything come of this?

**Comment by jsmedmar** [comment]
@gisellecory I didn't do it :/

**Comment by gisellecory** [comment]
Do you know more about the suggested approach above? I code in Python but those details are a bit lost on me and any guidance would be much appreciated!

**Comment by jsmedmar** [comment]
@gisellecory Im giving this a try at the moment, will update u soon. promise

**Comment by jsmedmar** [comment]
Found a very nice solution, will share it here tomorrow and also submit PR

**Comment by gisellecory** [comment]
Exciting - thanks!

**Comment by jsmedmar** [comment]
@gisellecory, you can try this as for now

```python
from PIL import Image
from PIL import ImageFilter
from wordcloud import WordCloud
from wordcloud import STOPWORDS
import numpy as np


def draw_contour(img, mask, contour_width, contour_color):
    """Draw mask contour on a pillow image."""
    contour = Image.fromarray(mask)
    contour = contour.resize(img.size)
    contour = contour.filter(ImageFilter.FIND_EDGES)
    contour = np.array(contour)

    # make sure borders are not drawn
    contour[[0, -1], :] = 0
    contour[:, [0, -1]] = 0

    # use a gaussian to define the contour width
    radius = contour_width / 10
    contour = Image.fromarray(contour)
    contour = contour.filter(ImageFilter.GaussianBlur(radius=radius))
    contour = np.array(contour) > 0
    contour = np.dstack((contour, contour, contour))

    # color the contour
    ret = np.array(img) * np.invert(contour)
    if contour_color != 'black':
        color = Image.new(img.mode, img.size, contour_color)
        ret += np.array(color) * contour

    return Image.fromarray(ret)


mask = np.array(Image.open("alice_mask.png"))
text = open("alice.txt").read()

wc = WordCloud(
    background_color="white",
    max_words=2000,
    mask=mask,
    stopwords=STOPWORDS,
    )

wc.generate(text)
contour = draw_contour(wc.to_image(), wc.mask, 1, 'blue')
contour.show()
```

**Comment by gisellecory** [comment]
I realise I never thanked you for this @jsmedmar - thank you!!

**Comment by santsang** [comment]
I spent hours in adding contour.  Python hanged at:

>    ret += np.array(color) * contour

I got the same error issue regardless I ran the suggested script or the native one (packed with wordcloud).  Perhaps I have to use packages additional to the suggested one?  Thanks for your suggestions.

**Comment by jsmedmar** [comment]
@santsang What about trying with these updates: https://github.com/amueller/word_cloud/pull/441/files

**Comment by santsang** [comment]
Many thanks, Juan, for your advice.  Not 100% sure how does it work.  Would you please give an example?

BTW, I find using a colour mask is better than a BW mask.  The shape of the word cloud is much more obvious for the former.

**Comment by amueller** [comment]
Well you can just set ``contour_width=`` a number bigger than 0 and that should work.

**Comment by santsang** [comment]
Thanks for your reply.  Yes, I knew that.  What I meant was that do I need to download that file and run all the functions?

BTW, I used your package for my reprint.  I have cited your package.

Sandro

**Comment by amueller** [comment]
It might mean you have an old version. What is ``wordcloud.__version__``?

**Comment by santsang** [comment]
Hi Andreas

Thank you for your suggestion.

I run WordCloud in the colab environment.  Following your suggestion, I installed 1.6.0 version.  Here are the info:

Requirement already satisfied: setuptools in /usr/local/lib/python3.6/dist-packages (from kiwisolver>=1.0.1->matplotlib->wordcloud==1.6.0) (45.1.0)
...
---> 92 plt.imshow(wordcloud, interpolation="bilinear")
...
/usr/local/lib/python3.6/dist-packages/wordcloud/wordcloud.py in _draw_contour(self, img)
    767 
    768         # color the contour
--> 769         ret = np.array(img) * np.invert(contour)

I did import numpy before I run the script.  In any case, I am happy with the diagram without contour.  I just want to give the info that can potentially help the development.

All the best

Sandro
